tensorflow_encrypted/tensor/native.py

```diff
@@ -90,6 +95,28 @@ def strided_slice(self, args: Any, kwargs: Any) -> 'NativeTensor':
     def reshape(self, axes: Union[tf.Tensor, List[int]]) -> 'NativeTensor':
         return NativeTensor(tf.reshape(self.value, axes), self.modulus)
 
+    def binarize(self) -> 'NativeTensor':
+        """Computes bit decomposition of tensor
+         tensor: ndarray of shape (x0, ..., xn)
+        returns: a binary tensor of shape (x0, ..., xn, bits) equivalent to tensor
+        """
+        if isinstance(self.value, np.ndarray):
+            bitwidths = np.arange(bits, dtype=np.int64)
+            for i in range(len(self.shape)):
+                bitwidths = np.expand_dims(bitwidths, 0)
+            val = np.expand_dims(self.value, -1)
+            val = np.right_shift(val, bitwidths) & 1
+
+            return NativeTensor.from_native(val, p)
+        elif isinstance(self.value, tf.Tensor):
+            bitwidths = np.arange(bits, dtype=np.int64)
```

Comment:
should this be the `tf.range`?

Reply:
numpy does work here and below but will change it to tf to be consistent!